### PR TITLE
Ccp 1646 spike create prototype for fetching

### PR DESCRIPTION
**Pull request by aaron-martin**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: extension/utilities/PluginApi.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  /**
+   * Sanitizes params for use in logging
+   * @param {*} params Params to be logged
+   * @return {*|{}}
+   */
+  sanitizeForLogging (params) {
+    let sanitizedForLoggingBody = params
+    if (typeof params === 'object') {
```
Comment (Carsten89) [inline extension/utilities/PluginApi.js]: not sure if it is relevant here, but keep in mind:
null is also an object `typeof null === 'object'`

File: extension/utilities/PluginApi.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      trace_id: 'sma-9412'
+    }
+    return this.call(form)
+  }
+
+  async checkCart (cart) {
+    const form = {
+      action: 'check_cart',
+      shop_number: this.shopNumber,
+      trace_id: 'sma-2149',
```
Comment (Carsten89) [inline extension/utilities/PluginApi.js]: where does `sma-2149` come from? why 2149

File: frontend/actions/index.js [hunk 3]
```diff
@@ -50,16 +50,13 @@ export const fetchBoltCartToken = () => (dispatch, getState) => {
...
-      .setInput({ orderId: 'dummy' })
-      .dispatch();
+    await new PipelineRequest('shopgate-project.bolt.clearCart').dispatch();
   } catch (err) {
     logger.error(err);
   }
 
   const products = getCartProducts(getState());
-  const order = formatTransaction(transaction, products);
+  const order = formatTransaction(transaction, products)
```
Comment (Carsten89) [inline frontend/actions/index.js]: please keep the `;` in the frontend code

File: frontend/portals/CartCheckoutButton/connector.js [hunk 4]
```diff
@@ -9,10 +9,15 @@ const mapStateToProps = state => ({
...
+ * @param {Function} dispatch The store dispatcher.
+ * @return {Object} The extended component props.
+ */
+const mapDispatchToProps = dispatch => ({
   fetchBoltCartToken,
-  processOrder,
-  initiatedCheckout,
-};
+  processOrder: transaction => dispatch(processOrder(transaction)),
+  initiatedCheckout: () => dispatch(initiatedCheckout()),
```
Comment (Carsten89) [inline frontend/portals/CartCheckoutButton/connector.js:20]:
```suggestion
  initiatedCheckout,
```

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: extension/utilities/PluginApi.js (hunk 1 above)
Comment (aaron-martin) [reply extension/utilities/PluginApi.js]: added check for the truthyness of the params in addition to its type.

File: frontend/actions/index.js (hunk 3 above)
Comment (aaron-martin) [reply frontend/actions/index.js]: oops. I put the semi colon back.

File: frontend/portals/CartCheckoutButton/connector.js (hunk 4 above)
Comment (aaron-martin) [reply frontend/portals/CartCheckoutButton/connector.js:20]: I tested this a few different ways. It seems that initiatedCheckout must be passed to dispatch for it to be called.

File: extension/utilities/PluginApi.js (hunk 2 above)
Comment (aaron-martin) [reply extension/utilities/PluginApi.js]: This was copied from some documentation or a sample log fie. It is meant to be a unique id made per call. I added a way to generate the trace id.

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**